rustfmt-config/src/lib.rs

```diff
@@ -48,8 +48,7 @@ create_config! {
     tab_spaces: usize, 4, true, "Number of spaces per tab";
     newline_style: NewlineStyle, NewlineStyle::Unix, true, "Unix or Windows line endings";
     indent_style: IndentStyle, IndentStyle::Block, false, "How do we indent expressions or items.";
-    use_small_heuristics: bool, true, false, "Whether to use different formatting for items and\
-        expressions if they satisfy a heuristic notion of 'small'.";
+    use_small_heuristics: LineWidth, LineWidth::Medium, false, "The general width of lines.";
```

Comment:
I think it is better to talk about how aggressively we put items across multiple lines, rather than 'general width of lines', which seems to invite confusion with `max_width`. So, rather than LineWidth::Minimum, we talk about making items as vertically formatted as possible, whereas Maximum is about making items maximally horizontal. (Or mutli-line vs single-line, if you prefer)

Reply:
This is still the biggest thing holding me back from finishing up this PR. The line length heuristics are confusing. I don't think "more horizontal" vs "more vertical" even really describes the options.

#### `use_small_heuristics: true`: (default)

- function calls: 60%
- struct literals: 18%
- struct variants: 35%
- arrays: 60%
- chains: 60%
- if/else: 50%

#### `use_small_heuristics: false`

- function calls: 100%
- struct literals: 0%
- struct variants: 0%
- arrays: 100%
- chains: 100%
- if/else: 0%

This means that enabling heuristics leads to code being sometimes more vertical and sometimes less vertical. This makes it really hard for me to describe what `use_small_heuristics: false` does in plain English, which also makes it hard to come up with a non-boolean name for what it does :(

Reply:
Hmm, yeah I know what you mean. The logic is that with `use_small_heuristics` false, we use the same formatting for small things as we do for large things, whereas when true we do things differently depending on the size - we've never tried to frame it as making lines longer or shorted, just 'better' (easier to read, prettier, etc). I guess the trouble with describing this is that even without `use_small_heuristics`, the set of things which is put on one line, rather than multiple lines is somewhat arbitrary.

Perhaps values like `None` (today's false), `Default` (today's true), and `Max` (for the new stuff) is OK. They're not very self-explanatory, but I don't think we'll do better.